runbotics/runbotics-scheduler/src/websocket/bot/bot.gateway.ts

```diff
@@ -124,4 +123,34 @@ export class BotWebSocketGateway implements OnGatewayDisconnect, OnGatewayConnec
         await this.botProcessEventService.updateProcessInstanceEvent(processInstanceEvent, bot);
         this.logger.log(`<= Success: process-instance-event (${processInstanceEvent.executionId}) updated by bot (${installationId}) | step: ${processInstanceEvent.step}, status: ${processInstanceEvent.status}`);
     }
+
+    private async changeEventStatusOnInstanceTermination(bot: IBot, processInstance: IProcessInstance, retryCount = 0) {
+        if(retryCount === 3) {
+            this.logger.log(`<= No active events of process-instance (${processInstance.id}) were found. Aborting.`);
+            return;
+        }
+
+        const activeEvents: IProcessInstanceEvent[] = await this.processInstanceEventService.findActiveByProcessInstanceId(processInstance.id);
+        
+        if(activeEvents.length === 0) {
+            this.logger.log(`<= No active events of process-instance (${processInstance.id}) were found. Retrying (${retryCount + 1}/3)`);
+            setTimeout(() => this.changeEventStatusOnInstanceTermination(bot, processInstance, retryCount + 1), 1000);
+            return;
+        }
+
+        await activeEvents.forEach(async (event) => {
+            const newProcessInstanceEvent: IProcessInstanceEvent = { 
+                ...event, 
+                status: ProcessInstanceEventStatus.TERMINATED, 
+                processInstance: {
+                    id: processInstance.id,
+                    process: {
+                        id: processInstance.process.id
+                    }
+                }
+            };
+
+            await this.updateProcessInstanceEvent(bot, newProcessInstanceEvent);
+        });
+    }
```

Comment:
this method should be in one of the services. Treat the gateway class like a controller class, business logic should be moved to another layer such as a service